src/content/BatchResponseContent.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module BatchResponseContent
+ */
+
+import { BatchResponseBody, BatchResponse } from "./BatchItem";
+import { Parsable, deserializeFromJson, ParsableFactory } from "@microsoft/kiota-abstractions";
+
+/**
+ * @class
+ * Class that handles BatchResponseContent
+ */
+export class BatchResponseContent {
+  /**
+   * To hold the responses
+   */
+  private readonly responses: Map<string, BatchResponse>;
+
+  /**
+   * @public
+   * @constructor
+   * Creates the BatchResponseContent instance
+   * @param {BatchResponseBody} response - The response body returned for batch request from server
+   * @returns An instance of a BatchResponseContent
+   */
+  public constructor(response: BatchResponseBody) {
+    this.responses = new Map();
+    this.update(response);
+  }
+
+  /**
+   * @public
+   * Updates the Batch response content instance with given responses.
+   * @param {BatchResponseBody} response - The response json representing batch response message
+   * @returns Nothing
+   */
+  public update(response: BatchResponseBody): void {
```

Comment:
this can be private?
```suggestion
  private update(response: BatchResponseBody): void {
```